superset/utils/core.py

```diff
@@ -349,6 +349,23 @@ def datetime_f(dttm):
     return "<nobr>{}</nobr>".format(dttm)
 
 
+def timedelta_f(td):
+    """
+    Ensures negative time deltas are easily interpreted by humans
+
+    >>> td = timedelta(0) - timedelta(days=1, hours=5,minutes=6)
+    >>> str(td)
+    '-2 days, 18:54:00'
+    >>> timedelta_f(td)
+    '-1 day, 5:06:00'
+    """
+    if td < timedelta(0):
+        return "-" + timedelta_f(-td)
```

Comment:
This recursion feels slightly unnecessary. Perhaps just doing `return "-"+ str(abs(td))` would be simpler?

Reply:
That's indeed simpler, I thought the recursion did more than it actually did. Simplified now!